spec/models/claim/advocate_claim_spec.rb

```diff
@@ -87,36 +87,36 @@
     end
   end
 
-  context 'eligible fee types' do
+  describe 'eligible fee types' do
     subject(:claim) { build(:unpersisted_claim) }
 
-    before(:all) do
-      @bft1 = create(:basic_fee_type, roles: %w[agfs agfs_scheme_9 agfs_scheme_10], description: 'bft1')
-      @bft2 = create(:basic_fee_type, :lgfs, description: 'bft2')
-      @bft3 = create(:basic_fee_type, description: 'bft3')
-      @bft4 = create(:basic_fee_type, roles: %w[agfs agfs_scheme_9], description: 'bft4')
-      @bft5 = create(:basic_fee_type, roles: %w[agfs agfs_scheme_10], description: 'bft5')
-      @mft1 = create(:misc_fee_type, :agfs_scheme_9)
-      @mft2 = create(:misc_fee_type, :lgfs)
-      @mft3 = create(:misc_fee_type, :agfs_scheme_10)
-      @fft1 = create(:fixed_fee_type)
-      @fft2 = create(:fixed_fee_type, :lgfs)
-    end
+    let(:basic_fee_type_schemes_nine_and_ten) { create(:basic_fee_type, roles: %w[agfs agfs_scheme_9 agfs_scheme_10]) }
+    let(:basic_fee_type_no_scheme) { create(:basic_fee_type) }
+    let(:basic_fee_type_scheme_nine) { create(:basic_fee_type, roles: %w[agfs agfs_scheme_9]) }
+    let(:basic_fee_type_scheme_ten) { create(:basic_fee_type, roles: %w[agfs agfs_scheme_10]) }
```

Comment:
I cannot see these claims being created before the tests are run so I'm a little suspicious about the tests if they are still passing. Am I missing something?

Reply:
This is creating fee types rather than claims. It sets up a number of fee types which apply to various fee schemes. The tests below then call `eligible_basic_fee_types` on a `claim` (which is created just before those `let`s) for a given fee scheme and checks that the correct fee types are returned.

Originally the setup was all done in the single `before(:all)` block but I've changed it so that the four fee types which are explicitly checked for in the tests are created in `let` statements (with clearer names, eg `@bft1` is now `basic_fee_type_schemes_nine_and_ten`), while the six which aren't referenced are created in the `before` (arguably we could delete them all together - they just exist to prove that, for example, an LGFS fee type isn't included on an AGFS claim).